src/backend/distributed/commands/table.c

```diff
@@ -1133,17 +1142,35 @@ ErrorIfUnsupportedAlterTableStmt(AlterTableStmt *alterTableStatement)
 				break;
 			}
 
-			case AT_SetNotNull:
+			case AT_EnableTrig:
+			case AT_EnableAlwaysTrig:
+			case AT_EnableReplicaTrig:
+			case AT_EnableTrigUser:
+			case AT_DisableTrig:
+			case AT_DisableTrigUser:
 			case AT_EnableTrigAll:
 			case AT_DisableTrigAll:
+			{
+				if (!IsCitusLocalTable(relationId))
+				{
+					const char *relationName = alterTableStatement->relation->relname;
+					ereport(ERROR, (errmsg("cannot execute ALTER TABLE ENABLE/DISABLE "
+										   "TRIGGER command for relation \"%s\" because "
+										   "it is either a distributed table or a "
```

Comment:
For example, user executed the command for a reference table and we say that  it is supported for citus local tables, I don't think it would make much sense to users. What do you think ?

Reply:
This is more of a general discussion maybe, but in this example, how will user know what to do? If the error message says that it is only supported for citus local tables, then:
- they know why the command failed, because their table was not `citus local table` type
- they know what they should do to not get the error
- (Minor)The error message doesn't include any negative words in it

However, when we say that "X is not supported", some questions are still left unanswered:
- Is this never supported? 
- If this is supported, what supports it? (For example in this example users won't know all the table types in citus)

I think a good error message should be direct about what is expected, what is the solution etc with possibly zero negative words if possible.

We can possibly create an issue on this as well (and discuss it) because errors are pretty important for users I think

Reply:
> how will user know what to do?

We already don't support triggers on other citus table, there is no point for them to execute these commands.

> what supports it? (For example in this example users won't know all the table types in citus)

To me, knowing which type of citus tables support this command is not beneficial. For example, I want to create & process triggers on a distributed table but it is not a solution for me to make my table a a citus local table

Reply:
>To me, knowing which type of citus tables support this command is not beneficial. For example, I want to create & process triggers on a distributed table but it is not a solution for me to make my table a a citus local table

I get your point, but on the other hand, I find it beneficial to know what the solution is even though I won't convert my table from a distributed to a citus local table. Again this is more of a general discussion maybe we can get input from other people as well @marcocitus, @onderkalaci ?

Reply:
I think the "only supported for local tables" wording makes sense to me.